utils/service_discovery/abstract_config_store.py

```diff
@@ -152,11 +244,12 @@ def get_checks_to_refresh(self, identifier, **kwargs):
         if to_check:
             return to_check
         else:
-            # lastly fallback to auto_conf
-            return set(self.identifier_to_checks[self._get_image_ident(identifier)])
+            # lastly fallback to legacy name for auto-conf
+            return self.template_cache.get_check_names(self._get_image_ident(identifier))
```

Comment:
Question: can we have checks under the legacy name that should also be applied? So a case where `identifier` and the legacy `self._get_image_ident(identifier)` would both apply to an image?